SPECIFICATION.md

```diff
@@ -0,0 +1,177 @@
+> n is four to six hexadecimal digits, using the digits 0–9 and uppercase
+> letters A–F (for 10 through 15, respectively).
+>
+> Leading zeros are omitted, unless the code point would have fewer than four
+> hexadecimal digits—for example, U+0001, U+0012, U+0123, U+1234, U+12345,
+> U+102345.
+>
+> The U+ may be omitted for brevity in tables or when denoting ranges.
+
+The glyph declaration is followed by a list of polylines or glyph references,
+each on a single line.
+
+#### Polylines
+
+A polyline looks like this:
+
+    0,0;3,9;6,0
+
+It consists of coordinate pairs, separated by semicolons (`;`). The X and Y
+coordinates are separated with a comma (`,`).  The coordinates are arbitrary
+precision floating point numbers using the full stop (U+002E) as decimal mark.
+If a coordinate is a whole number, the decimal mark and the fractional part of
+the number may be omitted.
+
+The coordinate system starts at the lower left corner. Positive X coordinates
+point to the right. Positive Y coordinates point to the top. Negative
+coordinates are not allowed.
+
+          Y
+    (0,Y) +
+          |
+          |
+          +-----+ X
+      (0,0)   (X,0)
+
+The example above would draw the pointy "outer frame" of the character "A".
+Here is the complete definition of the glyph for the character "LATIN CAPITAL
+LETTER A (U+0041)":
+
+    [0041] A
+    0.8333,2.5;5.1666,2.5
+    0,0;3,9;6,0
+
+#### References
+
+Because many glyphs are very similar, references may be used as "includes".
+References start with the "C" symbol (U+0043), followed by a codepoint. Here is
+the definition of the glyph for the character "LATIN CAPITAL LETTER A WITH
+CIRCUMFLEX (U+00C2)":
+
+    [00c2] Â
+    C0041
+    1.5,11.5;3,13;4.5,11.5
+
+An implementation would simply replace the "C0041" part with the expanded
+polylines from the referenced glyph.
+
+Multiple references can be used in a single glyph definition, but they must all
+be on their own line.
+
+To prevent reference loops and to facilitate single-pass parsers, only backward
+references are allowed. All references glyphs must have been previously defined.
+
+
+## Format Versioning
+
+The FontoBene format follows [Semantic Versioning version 2][semver-2].
+
+
+## Formal Specification
+
+### Grammar (PEG)
+
+TODO
+
+### Standardized Headers
+
+#### [format]
+
+| Name | Description | Count | Example |
+| --- | --- | --- | --- |
+| format | The string "FontoBene". | 1 | `FontoBene` |
+| format_version | The version of the FontoBene format that this file adheres to. | 1 | `1.1.2` |
+
+#### [font]
+
+| Name | Description | Count | Example |
+| --- | --- | --- | --- |
+| name | The name of this font. | 1 | `LibrePCB Font` |
+| id | The identifier of this font. MUST only contain lowercase letters and minus characters. MUST start and end with a lowercase letter. | 1 | `librepcb` |
+| version | The version of this font. SHOULD follow semantic versioning. | 0-1 | `0.4.1` |
```

Comment:
Shouldn't the version number be mandatory? I think the version is quite important, and if it's not used one can just write "1.0" into this field ;)